builder/clique_builder.go

```diff
@@ -0,0 +1,398 @@
+package builder
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"sync"
+	"time"
+
+	bellatrixapi "github.com/attestantio/go-builder-client/api/bellatrix"
+	capellaapi "github.com/attestantio/go-builder-client/api/capella"
+	apiv1 "github.com/attestantio/go-builder-client/api/v1"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	blockvalidation "github.com/ethereum/go-ethereum/eth/block-validation"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/flashbotsextra"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/flashbots/go-boost-utils/bls"
+	"github.com/flashbots/go-boost-utils/ssz"
+	"github.com/flashbots/go-boost-utils/utils"
+	"github.com/holiman/uint256"
+	"golang.org/x/time/rate"
+)
+
+type CliqueBuilder struct {
+	ds        flashbotsextra.IDatabaseService
```

Comment:
I like it to be little bit descriptive
```suggestion
	dbService        flashbotsextra.IDatabaseService
```